tools/wheel_resolver/src/wheel_tags/tags.py

```diff
@@ -0,0 +1,128 @@
+"""
+Some methods for wheel fetching and selection
+"""
+
+import os
+from tools.wheel_resolver.third_party.python.packaging.utils import parse_wheel_filename
+import tools.wheel_resolver.third_party.python.packaging.tags as tags
+import tools.wheel_resolver.third_party.python.distlib.locators as locators
+import third_party.python.setuptools as pkg
+
+
+def is_compatible(wheel, taglist):
+    """
+    Check whether the given python package is a wheel compatible with the
+    current platform and python interpreter.
+    Compatibility is based on https://www.python.org/dev/peps/pep-0425/
+    """
+    # Get the tag from the wheel we're checking
+    _, _, _, tag = parse_wheel_filename(wheel)
+
+    # taglist is a list of tags that we've got either from the user
+    # or we've auto-generated them for this system
+    if taglist is None:
+        taglist = tags.sys_tags()
+
+    for system_tag in taglist:
+        if system_tag in tag:
+            return True
+    return False
+
+
+def get_basename(url):
+    return os.path.basename(url)
+
+
+def is_wheel_file(url):
+    """ Return whether or not this is a .whl file """
+    basename = os.path.basename(url)
+    _, ext = os.path.splitext(basename)
+    return ext == '.whl'
+
+
+def generate_tags_from_all_archs(archs):
+    result = []
+    for arch in archs:
+        result += list(tags.generic_tags(interpreter=None,
+                                         abis=None,
+                                         platforms=[arch]))
+        result += list(tags.compatible_tags(python_version=None,
+                                            interpreter=None,
+                                            platforms=[arch]))
+
+    return result
+
+
+def get_url(urls, archs):
+    """
+    From the list of urls we got from the wheel index, return the first
+    one that is compatible (either with our system or a provided one)
+    """
+    taglist = None
+    if len(archs) != 0:
+        # Make a list of tags
+        taglist = generate_tags_from_all_archs(archs)
+        # Just check that we've got some tags
+        if len(taglist) == 0:
+            print("Didn't generate any tags for arch list:", archs)
+            return 1
```

Comment:
I've seen this pattern being used around, which reminds of my good old C days :-) In python, we seem to use zero/null values instead.